bindings/java/java/src/test/java/org/ethereum/evmc/EvmcTest.java

```diff
@@ -230,16 +255,21 @@ void testExecute_EVMC_CREATE() throws Exception {
       char[] sender = "39bf71de1b7d7be3b51\\0".toCharArray();
       char[] recipient = "53cf77204eEef952e25\0".toCharArray();
       char[] value = "1\0".toCharArray();
-      char[] inputData = "hello w\0".toCharArray();
+      char[] inputDataPlaceholder = {};
+      byte[] inputDataBytes = {0x68, 0x65, 0x6c, 0x6c, 0x6f};
+      ByteBuffer inputDataByteBuffer =
+          ByteBuffer.wrap(inputDataBytes).order(ByteOrder.nativeOrder());
       long gas = 200000;
       int depth = 0;
       ByteBuffer msg =
-          new TestMessage(kind, sender, recipient, value, inputData, gas, depth).toByteBuffer();
+          new TestMessage(kind, sender, recipient, value, inputDataPlaceholder, gas, depth)
```

Comment:
I was wondering what this `TestMessage` is and why it needs the `inputDataPlaceholder`.

It turns out `vm.execute` still gets this entire data structure with `input_data`/`input_length`.

This is a bad idea having shadow data which is not checked. Probably the better option is removing those from the `msg` structure, which also means it needs to be reassambled in the binding.